lib/assessment.js

```diff
@@ -24,22 +28,51 @@ var sum = function sum() {
 // if invoking the callback for any element
 // in the array returns a truthy value
 var any = function any(array, callback, thisArg) {
-
+  callback(currentValue, index, array)
+  var result = false;
+  array.any(function(currentValue, index, array){
+    result = result || callback(currentValue, index, array);
+  });
+  return result;
 };
 
-// Question 4
+
+// var every = function every(array, callback, thisArg) {
+//   // callback(currentValue, index, array)
+//   var result = true;
+//   array.forEach(function(currentValue, index, array) {
+//     // if callback ever returns false, result will be set
+//     // to false amd the callback will not be called in any
+//     // subsequent iterations.
+//     result = result && callback(currentValue, index, array);
+//   });
+
+
+
+/// Question 4
 // Write a function that returns a new array
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 var select = function select(array, callback, thisArg) {
+  var result = [];
+  array.forEach(function(currentValue, index, array) {
+    result = result && callback(currentValue, index, array);
+  });
 
+  return result;
+};
 };
 
 // Question 5
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
 var find = function find(array, callback, thisArg) {
-
+ callback(currentValue, index, array)
+  var result = false;
+  array.any(function(currentValue, index, array){
+    result = result || callback(currentValue, index, array);
+  });
+  return result;
```

Comment:
Not quite.  
```javascript
// Write a function that returns the first element
// for which invoking the callback returns a truthy value	
```
We want to iterate through the entire array and save the first value for which `callback(value)` is true. Think you can figure out how to implement it?